cmd/fs-v1-metadata.go

```diff
@@ -315,16 +316,23 @@ func createFormatFS(fsPath string) error {
 }
 
 func initFormatFS(fsPath string) (rlk *lock.RLockedFile, err error) {
-	// This loop validates format.json by holding a read lock and
-	// proceeds if disk unformatted to hold non-blocking WriteLock
-	// If for some reason non-blocking WriteLock fails and the error
-	// is lock.ErrAlreadyLocked i.e some other process is holding a
-	// lock we retry in the loop again.
 	for {
+		printedCh := delayPrint("Waiting to acquire lock.. ", time.Second)
+
 		// Validate the `format.json` for expected values.
 		rlk, err = checkLockedValidFormatFS(fsPath)
+
+		// Waits on delayed printer to send back notification
+		// if the previous message was indeed printed.
+		printed := <-printedCh
 		switch {
 		case err == nil:
+			// Print this message only if previously
+			// read-lock was waiting and blocked.
+			if printed {
```

Comment:
Even in this implementation, I think there is a bug. The readlock may block when a write lock is acquired by someone else, but if a write lock is attempted, there are sleep statements and formatting actions that could take a while - in this case the message needs to be printed. So the `printedCh` should be listened on only in the `err == nil` case. In the current implementation, the sleep-after-writelock-attempt and formatting steps might take time too, but the message would not be printed.